tools/scripts/artifacts/nx-project.ts

```diff
@@ -159,16 +160,27 @@ export class NxProject {
     console.log(...snapshots);
     for (const snapshot of snapshots) {
       await this.deleteArtifact(
-        jfrogCredentials,
         Utils.getVersionFromSnapshotString(snapshot)
       );
     }
   }
 
-  public async deleteArtifact(
-    jfrogCredentials: JfrogCredentials,
-    version: Version
-  ) {
+  private packageExists(pkg: string, version: string) {
+    const scopeSearchResult = execSync(
+      `npm search ${pkg} --json`
+    ).toString();
+    console.log('scopeSearchResult: ', scopeSearchResult);
+    const npmSearchResults: NpmPackage[] = JSON.parse(scopeSearchResult);
+    return npmSearchResults.some((entry) => entry.name === pkg && entry.versions.includes(version));
+  }
+
+  public async deleteArtifact(version: Version) {
+    console.log("Checking if package exists in registry");
+    if(!this.packageExists(`${this.scope}/${this.name}`, version.toString())) {
```

Comment:
Optional:

```suggestion
private scopedPackageName = `${this.scope}/${this.name}`;
//...

    if(!this.packageExists(scopedPackageName, version.toString())) {
```
Maybe introducing a variable like this can improve code quality

Reply:
Good tip. Updated

Reply:
That is what I was thinking, but I figured you should update every instance of `${this.scope}/${this.name}` in the entire file since there is more than just this function 👀